datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -81,11 +83,13 @@ pub struct ParquetExec {
     pruning_predicate: Option<PruningPredicate>,
     /// Optional hint for the size of the parquet metadata
     metadata_size_hint: Option<usize>,
+    /// Optional user defined parquet file reader factory
+    parquet_file_reader_factory: Option<Arc<dyn ParquetFileReaderFactory>>,
 }
 
 /// Stores metrics about the parquet execution for a particular parquet file
 #[derive(Debug, Clone)]
-struct ParquetFileMetrics {
+pub struct ParquetFileMetrics {
```

Comment:
It would be good if we could find a way to not make this public, as we're likely to want to alter it over time